sessions/workflow_dacx.go

```diff
@@ -0,0 +1,90 @@
+
+Pass an instance of `workflow.Context` and [`SessionOptions`](https://pkg.go.dev/go.temporal.io/sdk/workflow#SessionOptions) to the `CreateSession` API call and get a Session Context which contains metadata information of the Session.
+
+Use the Session Context to spawn all Activity Executions that should belong to the Session.
+All associated Activity Tasks are then processed by the same Worker Entity.
+*/
+
+type FileProcessingWFParam struct {
+	CloudFileLocation string
+}
+
+// SomeFileProcessingWorkflow is a Workflow Definition
+func SomeFileProcessingWorkflow(ctx workflow.Context, param FileProcessingWFParam) error {
+	activityOptions := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Minute,
+	}
+	ctx = workflow.WithActivityOptions(ctx, activityOptions)
+	// By placing all three steps in another function, the entire sequence retries on a different host from the first Activity on any error
+	// In a real application it might be reasonable to retry individual Activities as well as the whole sequence discriminating between different types of Errors.
+	// See the retryactivity sample for a more sophisticated retry implementation.
+	// https://github.com/temporalio/samples-go/tree/main/retryactivity
+	err := processFile(ctx, param)
+	return err
+}
+
+func processFile(ctx workflow.Context, param FileProcessingWFParam) error {
+	// Set the required timeouts for the Worker Session
+	sessionOptions := &workflow.SessionOptions{
+		CreationTimeout:  time.Minute,
+		ExecutionTimeout: time.Minute,
+	}
+	// Create a Session with the Worker so that all Activities execute with the same Worker.
+	sessionCtx, err := workflow.CreateSession(ctx, sessionOptions)
+	if err != nil {
+		return err
+	}
+	defer workflow.CompleteSession(sessionCtx)
+
+	var a *FileActivities
+	var downloadResult FileActivityResult
+	err = workflow.ExecuteActivity(sessionCtx, a.DownloadFile, param).Get(sessionCtx, &downloadResult)
+	if err != nil {
+		return err
+	}
+	processParam := FileActivityParam(downloadResult)
+	var processResult FileActivityResult
+	err = workflow.ExecuteActivity(sessionCtx, a.ProcessFile, processParam).Get(sessionCtx, &processResult)
+	if err != nil {
+		return err
+	}
+	uploadParam := FileActivityParam(processResult)
+	err = workflow.ExecuteActivity(sessionCtx, a.UploadFile, uploadParam).Get(sessionCtx, nil)
+	return err
+}
+
+/*
+When the `CreateSession` API is called, the Task Queue name that is specified in the `ActivityOptions` is used (or in the `StartWorkflowOptions` if the Task Queue name is not specified in `ActivityOptions`), and a Session is created with one of the Workers polling that Task Queue.
+
+The Session Context is cancelled if the Worker executing this Session dies or `CompleteSession()` is called.
+When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error may be returned if the Session framework detects that the Worker executing this Session has died.
```

Comment:
```suggestion
When using the returned Session Context to spawn Activity Executions, a `workflow.ErrSessionFailed` error is returned if the Session framework detects that the Worker executing this Session has died.
```
Or are there situations in which the `workflow.ErrSessionFailed` error isn't returned?